### ISSUE TITLE
podDefaults not being applied to notebook

### ISSUE THREAD

**Issue by nathanwong91** [kubeflow/kubeflow#5395]
/kind question

**Question:**
I'm trying to get podDefaults to mount a secret and an AWS iam role to notebooks that get created but I cannot seem to get them to work. 

I can see the labels when I describe the pod
```
add-kubeflow-role=true
add-github-secret=true
```
However they both aren't getting applied

```
Annotations:          kubernetes.io/psp: eks.privileged
                      sidecar.istio.io/status:
                        {"version":"1cdb312e0b39910b7401fa37c42113f6436e281598036cb126f9692adebf1545","initContainers":["istio-init"],"containers":["istio-proxy"]...
```
```
Mounts:
      /dev/shm from dshm (rw)
      /home/jovyan from workspace-test (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-editor-token-dpsjz (ro)
```
These are the podDefaults I'm adding

```
apiVersion: "kubeflow.org/v1alpha1"
kind: PodDefault
metadata:
  name: add-github-secret
spec:
  selector:
    matchLabels:
      add-github-secret: "true"
  desc: "add github credential"
  volumeMounts:
  - mountPath: /tmp/git
    name: credentials
  volumes:
  - name: credentials
    secret:
      secretName: github-credentials
```

```
apiVersion: "kubeflow.org/v1alpha1"
kind: PodDefault
metadata:
  name: add-kubeflow-role
spec:
  selector:
    matchLabels:
      add-kubeflow-role: "true"
    desc: "Assume kubeflow role"
    annotations:
      iam.amazonaws.com/role: role-name
```

Not sure if the podDefaults I'm using are wrong or something else is going on.

**Comment by waynehsei** [comment]
Are there any update of this? we ran to the same issue

**Comment by RDarrylR** [comment]
I had a similar problem.

I resolved it by updating the image for the deployment of the profile manager. There was some issue that an old version of this image was being referenced in the manifest when kubeflow was installed.

```
kubectl --record deployment.apps/profiles-deployment set image deployment.v1.apps/profiles-deployment manager=gcr.io/kubeflow-images-public/profile-controller:vmaster-ga49f658f -n kubeflow

kubectl rollout restart deployment profiles-deployment -n kubeflow
```

I found another issue for 1.1 below with a comment describing the above change. I am working with a KF 1.2 install so the above commands use the latest version of that image. If you are running 1.1 You likely want the version in the note below.

https://github.com/kubeflow/kubeflow/issues/5207#issuecomment-702341351
